### DISCUSSION TITLE
How to tar_load the path created by tar_render?

### DISCUSSION THREAD

**Discussion by JakeRuss** [ropensci/tarchetypes#58] [Help] [answered]
Every example I've been able to find of `tar_render()` has it at the end of that example's pipeline. However, I want to have additional `tar_target()`s after the `tar_render` step. How do I retrieve the file path returned from tar_render such that my targets post-tar_render will cue appropriately?

I have something that looks like, 
````
list(
tar_target(a, ...),
tar_target(b, ...),
tar_render(c, ...),
tar_target(d, write_to_s3_bucket(c)),
tar_target(e, alert_slack(d))
)
````
I can see that the path for `c` is recorded in `meta$path` as a vector (along with the path for the `.Rmd` file). How do I access that vector so that I can extract the first path and supply it to `write_to_s3_bucket()`? 

Currently, due entirely to my user error, targets is not recognizing `c` as a dependency of `d`, which allows `tar_make` to re-run `d` before re-running `c` and get stuck in an infinite loop.

**Comment by wlandau** [comment, marked as answer]
In normal usage, a `tar_render()` target returns a vector file paths involved in rendering: the source Rmd file, the output HTML (or pdf) file, and a `"_files"` directory path if it exists. So you can definitely do something like this in _targets.R:

```r
list(
  tar_render(report, "report.Rmd"),
  tar_target(s3, write_to_s3(grep(pattern = "index.html", x = report, value = TRUE, fixed = TRUE))
)
```

If you post a reprex of the specific issue, I would be happy to take a look. Not sure why your graph has trouble identifying dependencies.

**Reply by JakeRuss** [reply]
Your solution works great, Will, the graph is as I expected now. I was trying to do,
````
list(
  tar_render(report, "report.Rmd"),
  tar_target(s3, write_to_s3(report[1])) # assuming the desired path was first 
)
````
  I appreciate you explaining it even without a full reprex.
